src/components/application_manager/doc/SDL.SDD.StateController.dox

```diff
@@ -0,0 +1,73 @@
+
+*/
+
+/**
+\page state_controller_hight_level_logic 1 State Controller hight level logic
+
+## Introduction
+The state controller is responsible for keeping applications inappropriate HMI states 
+HMI state contains: 
+ - HMILevel 
+ - AudioStreamingState
+ - VideoStreamingState
+ - SystemContext 
+ 
+SDL should take care of conflicts between the HMI state of different apps. 
+For example, 2 apps cannot be in FULL at the same time. So if one application goes FULL, another one should be moved to BACKGROUND or LIMITED.
+Rules are described in the scope of the proposal https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0150-video-streaming-state.md 
+
+SDL should keep resolve application HMI states to avoid conflicts. There are several triggers for HMI state change:
+ - application activation
+ - application deactivation
+ - Start System Event from HMI (Phone call, TTS, etc ...)
+ - End System Event from HMI 
+ - Resumption
+
+\link application_manager::StateController 
+\endlink is responcible for this.
+
+Event from HMI should change HMI state for app registered applications.
+Changing rules described in  https://github.com/smartdevicelink/sdl_evolution/blob/proposals/0150-video-streaming-state.md 
+Some modifications introduced within https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0264-Separating-the-change-of-Audible-status-and-the-change-of-HMI-Status.md 
+
+Events from HMI are temporary and will be ended.
+After finishing of system Event all applications should receive a previous HMI state. Sp SDL should keep it.
+
+Multiple system events can present at the same time. SDL should calculate HMI state according to all of them.
+
+## Technical solution
+
+To track complicated logic of #application_manager::HmiState within system events SDL put events in internal "stack" of HMI States.
+
+There is a Regular HMI state of the application (STATE_ID_REGULAR).
+The regular state contains HMI Level, AudioStreamingState, VideoStreamingState, SystemContext of the app. 
+And regular hmi state is agnostic to system events. 
+A regular state can be changed by actiation\deactivation or resumption.
+
+There is a Temporary HMI state of the application.
+Temporary HMI state is not independent and should be on the top of Regular or another Temporary HMI state, consider it as parent state.
+Temporary HMI state does not contain any values, it contains the rules on how it changes values of the parent state. 
+The temporary state creates within SystemEvent start and deleted on SystemEvent end.
+
+SDL creates a linked list of HMI states for each app.
+This list contains at least one element: Regular HMIstate. And it always should be the first. 
+Temporary State added to this list on the system even start, and removed from this list on system event end. 
+
+System Events do not follow FILO, so events may be started on order 1, 2, 3, and ended on order 1, 3, 2.
+
+Stack system of application states: 
```

Comment:
@LuxoftAKutsan not sure why but I don't see these images in a generated document:
![image](https://user-images.githubusercontent.com/5848997/81309164-a0561180-9050-11ea-90eb-81204c7e718e.png)

Reply:
Probably because plant uml is not installed on your system